### DISCUSSION TITLE
API for instantiation and termination of edge servers after deployment

### DISCUSSION THREAD

**Discussion by alyhasansakr** [InterDigitalInc/AdvantEDGE#186] [Ideas]
**Problem description:**
For a resource limited multi-app environment (such as automotive case), edge (or fog) servers cannot be always running, instead they should be instantiated and terminated based on demand and available resources.

**Solution:**
API for creation and deletion of edge applications in active deployment. This allows an external orchestrator to manage that.

To be exact the following functions are needed:
- create edge app.
- delete edge app.

For creating a new edge app: input is exactly the same as you would provide when creating a new edge app using the front-end before deployment, and expected behavior is exactly the same as if you had created it using the front-end before deployment.

Deletion needs only the name of the edge app, and expected behavior is that the edge app is no longer running.

**Additional notes:**
The orchestrator can use the location service already available in the platform to determine where to create the new edge apps. Also, the orchestrator is responsible of resources definition and management.

**Comment by roymx** [comment]
Hello @alyhasansakr,

FYI - this feature finally made it to the top and should be available in [v1.7.0](https://github.com/InterDigitalInc/AdvantEDGE/wiki/Roadmap#project-progression) ~March/April timeframe
